### ISSUE TITLE
Broken savegame

### ISSUE THREAD

**Issue by Chance4us** [diasurgical/devilutionX#118]
For some reason my savegame got damaged. The file is attached and when loading the game crashes. The question now is if it is possible to find the reason and maybe also to repair the it?
[single_3.sv.zip](https://github.com/diasurgical/devilutionX/files/3213446/single_3.sv.zip)

**Comment by AJenbo** [comment]
I'm going to guess that you have been running both 32bit and 64bit builds, the 64bit builds save games are currently not compatible with the 32bit version. This is one of the remaining blocking issues for releasing 64bit builds: https://github.com/diasurgical/devilutionX/issues/53 and why they are considered experimental.

**Comment by Chance4us** [comment]
At the moment I am building only 32bit devilutionX because otherwise it is not possible to load/save a game. I think it could be related to an item I found in level 15 (hell). It was a long battle bow and I wondered about some fantastic values of it (+100% damage and lightning damage at the same time). Also it could be caused by the "veil of steel" helmet. So I hope we can find the reason by investigating maybe by using a trainer or in other way. I think it would be unfavorable to keep a such serious bug in devilution.

**Comment by Chance4us** [comment]
PS: I was just able to load the savegame into original diablo.
The picture is attached.
![sc](https://user-images.githubusercontent.com/45902449/58286377-e8801380-7dae-11e9-820a-66e5cc5170dd.png)

**Comment by Chance4us** [comment]
And now after playing with the broken one in original diablo little bit and saving once again the savegame seems to be repaired by the original game. It could be loaded in devilutionX again.
[single_3.sv.zip](https://github.com/diasurgical/devilutionX/files/3214239/single_3.sv.zip)

**Comment by AJenbo** [comment]
Please check things like this against http://www.lurkerlounge.com/diablo/jarulf/jarulf162.pdf so that we don't have to chase down hunches. Afaik that bow seams legit. 

Also when posting issues without a way to replicate the issue make sure to include as much information as possible, like what system are you running this on, what release/commit is it, did you build it your self, what toolchain was used. Was the game originally started on the same version.

What exactly does it mean when you say "playing with the broken one in original".

In short if we can't replicate the issue then we can't do anything about it. Save game can get corrupted in more then one way and once the save game contains the corrupted state the event has already happened so we only have the result not the cause.

**Comment by Chance4us** [comment]
I meant that the broken savegame could be loaded by Diablo.exe from CD of the manufacturer of the game in linux wine.

**Comment by Chance4us** [comment]
Additional information:
Operating System: Debian 10 (Testing) with SMP Debian 4.19.37-3 (2019-05-15) x86_64 GNU/Linux
Installed libraries: cmake g++-multilib libsdl2-mixer-dev:i386 libsdl2-ttf-dev:i386 libsodium-dev libsodium-dev:i386
Used build commands for devilutionX:
git clone https://github.com/diasurgical/devilutionX
cd devilutionX
mkdir build32
cd build32
cmake -DBINARY_RELEASE=ON -DASAN=OFF -DCMAKE_TOOLCHAIN_FILE=../CMake/32bit.cmake ..
linux32 make -j$(nproc)

Hardware specifications:
Mainboard: ASRock Fatal1ty AB350 Gaming-ITX/ac
CPU: AMD Ryzen 3 2200G with integrated GPU "AMD Radeon VEGA"

**Comment by Chance4us** [comment]
I don't know how to determine the release/commit of devilutionX but it is almost the latest build cloned 3 or 4 days ago.

**Comment by Chance4us** [comment]
Don't hesitate to contact me if more information is needed for investigation.

**Comment by AJenbo** [comment]
if you run `git log` you will see the latest commit that you have at the top of the list.

**Comment by Chance4us** [comment]
Thanks for fast feedback. git log file is attached immediately.
[gitlog.txt](https://github.com/diasurgical/devilutionX/files/3218023/gitlog.txt)

**Comment by Chance4us** [comment]
My biggest surprise is why Diablo.exe can handle the broken savegame and devilutionX not.

**Comment by AJenbo** [comment]
All we need is the commit hash :)
73a29f0e68b2ff9f23a319f747f45a905376ddf0
